File: src/test/java/com/icecreamqaq/test/db/dao/SisterDao.java
package com.icecreamqaq.test.db.dao;

import com.icecreamqaq.test.db.entity.Sister;
import com.icecreamqaq.yudb.jpa.JPADao;
import com.icecreamqaq.yudb.jpa.annotation.Select;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface SisterDao extends JPADao<Sister,Integer> {

    default void aaa(){
        System.out.println("This is Java fun aaa!");
    }

    @Select("from Sister where xxx = ?0")
    Sister getByXxx(String xxx);

    @NotNull
    List<Sister> findByName(@NotNull String s);

    @Nullable
    Sister findByNameAndId(@NotNull String s, int id);
}

File: src/test/kotlin/com/icecreamqaq/test/db/dao/SisterDao.kt
package com.icecreamqaq.test.db.dao

import com.icecreamqaq.test.db.entity.Sister
import com.icecreamqaq.yudb.YuDao
import com.icecreamqaq.yudb.jpa.annotation.Dao
import com.icecreamqaq.yudb.jpa.annotation.Execute
import com.icecreamqaq.yudb.jpa.annotation.Select

//class SisterDao : HibernateDao<Sister, Int>()

@Dao
interface SisterDao : YuDao<Sister, Int> {

    fun findByName(name: String): List<Sister>?

    fun findByNameAndId(name: String, id: Int): Sister

    @Execute("delete from Sister where name = ?")
    fun da(name: String)

    @Select("from Sister where name = ?")
    fun sa(name: String): Sister?

    @Select("from Sister")
    fun findAll(): List<Sister>

}